File: LibraryManagementSystem-backend/src/main/java/com/proiect/LibraryManagementSystem/model/Book.java
package com.proiect.LibraryManagementSystem.model;


import com.fasterxml.jackson.annotation.*;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import java.time.LocalDate;
import java.util.List;

@Entity
@Table(name = "books")
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
public class Book {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;
    private String title;

    @ManyToOne
    @JoinColumn(name = "author_id")
   // @JsonBackReference
    @JsonIgnoreProperties("books")
    private Author author;

    @JsonFormat(pattern = "dd.MM.yyyy", timezone = "UTC")
    private LocalDate publicationDate;

    private Integer quantity;

    private String category;

    private boolean active = true;

    @OneToMany(mappedBy = "book",cascade = CascadeType.ALL)
    @JsonIgnore
    //@JsonManagedReference(value = "book_loan")
    private List<BookLoan> loans;
}

File: client/client/models/Book.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using static System.Runtime.InteropServices.JavaScript.JSType;

namespace client.models
{
    public class Book
    {
        public int id { get; set; }
        public string title { get; set; }
        public Author author { get; set; }
        public string publicationDate { get; set; }
        public bool active { get; set; } = true;
        public int quantity { get; set; }  
        public string category { get; set; }

        public List<BookLoan> loans { get; set; } = new List<BookLoan>();
    }
}
